token/program2/src/instruction.rs

```diff
@@ -37,6 +37,8 @@ pub enum TokenInstruction {
         amount: u64,
         /// Number of base 10 digits to the right of the decimal place.
         decimals: u8,
+        /// Whether the mint freeze_authority should be enabled to freeze associated token accounts
+        can_freeze: bool,
```

Comment:
No freeze support for a token with fixed supply?

Reply:
That's supported, just have to pass an authority as the option 2nd account. I obv need to update the docs there.

Reply:
ah, got it.   the `process_initialize_mint()` logic based on the presence of the optional owner is a little complex, more so now.   I think that's going to be my main argument for dropping the owner as an account and passing the Owner/Freeze pubkey as optional fields to the instruction data

Reply:
Okay, I pushed what this would properly look like -- we need to independently declare mint fixed supply and/or can_freeze. But I hate it. I'll rework back on https://github.com/solana-labs/solana-program-library/issues/229#issuecomment-677865126 in the morn.